editor/editor_inspector.cpp

```diff
@@ -2543,10 +2543,13 @@ void EditorInspector::update_tree() {
 				while (script.is_valid()) {
 					StringName name = EditorNode::get_editor_data().script_class_get_name(script->get_path());
 					String icon_path = EditorNode::get_editor_data().script_class_get_icon_path(name);
-					if (name != StringName() && icon_path.length()) {
+					if (name != StringName() && !icon_path.is_empty()) {
 						category->icon = ResourceLoader::load(icon_path, "Texture");
 						break;
 					}
+					if (const EditorData::CustomType *ctype = EditorNode::get_editor_data().get_custom_type_by_path(script->get_path())) {
+						category->icon = ctype->icon;
+					}
```

Comment:
Does this fall under the forbidden craft of "assignment in conditions"? Guess that may need to be expanded into multiple lines.

Also, shouldn't it break on match?